src/include/common/cpu/rpp_cpu_rgb_hsv_conversion.hpp

```diff
@@ -0,0 +1,227 @@
+/*
+MIT License
+
+Copyright (c) 2019 - 2025 Advanced Micro Devices, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "rpp_cpu_common.hpp"
+
+#if __AVX2__
+
+// Converts RGB to HSV color space using AVX vectorization, processes 8 pixels simultaneously
+inline void rgb_to_hsv(__m256 &pVecR, __m256 &pVecG, __m256 &pVecB, __m256 &pH, __m256 &pS, __m256 &pV, __m256 &pAdd)
+{
+    __m256 pMask[4], pDelta;
+    // RGB to HSV
+    pV = _mm256_max_ps(pVecR, _mm256_max_ps(pVecG, pVecB));                                                         // cmax = RPPMAX3(rf, gf, bf);
+    pS = _mm256_min_ps(pVecR, _mm256_min_ps(pVecG, pVecB));                                                         // cmin = RPPMIN3(rf, gf, bf);
+    pDelta = _mm256_sub_ps(pV, pS);                                                                                 // delta = cmax - cmin;
+    pH = avx_p0;                                                                                                    // hue = 0.0f;
```

Comment:
pH and pS is getting reassigned in line #41 and #44. Why is it initialized here

Reply:
Fixed